tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcCommon.cs

```diff
@@ -0,0 +1,119 @@
+﻿// <copyright file="GrpcCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Grpc
+{
+    internal static class GrpcCommon
+    {
+        public const string OperationName = "grpc.request";
+        public const string ServiceName = "grpc-client";
+        public const string RequestMetadataTagPrefix = "grpc.request.metadata";
+        public const string ResponseMetadataTagPrefix = "grpc.response.metadata";
+
+        public static void AddGrpcTags(GrpcTags tags, Tracer tracer, int grpcType, string? name, string? path, string? serviceName)
+            => AddGrpcTags(tags, tracer, GetGrpcMethodKind(grpcType), name, path, serviceName);
+
+        public static void AddGrpcTags(GrpcTags tags, Tracer tracer, string grpcType, string? name, string? path, string? serviceName)
+        {
+            tags.MethodKind = grpcType;
+            tags.MethodName = name;
+            tags.MethodPath = path;
+
+            if (!string.IsNullOrEmpty(serviceName))
+            {
+                // get the package and service name
+                var indexOf = serviceName!.LastIndexOf('.');
+
+                if (indexOf > 0 && indexOf < serviceName.Length - 2)
+                {
+                    tags.MethodPackage = serviceName.Substring(startIndex: 0, length: indexOf);
+                    tags.MethodService = serviceName.Substring(startIndex: indexOf + 1);
+                }
+                else
+                {
+                    tags.MethodService = serviceName;
+                }
+            }
+
+            tags.SetAnalyticsSampleRate(IntegrationId.Grpc, tracer.Settings, enabledWithGlobalSetting: false);
```

Comment:
It is common for the gRPC server spans to be a root span? If so, we may want to use `enabledWithGlobalSetting=true`. Unless the RFC says otherwise in which case you can ignore and resolve this comment :)

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2545/commits/51db656a7f8e552263a1d3ebd60b13e6663b5f2a. 

Also, I didn't realise that was the purpose of `enabledWithGlobalSetting`, so thanks for pointing it out!